src/settings.gradle

```diff
@@ -1,7 +1,15 @@
 pluginManagement {
-  plugins {
-    id 'com.github.johnrengelman.shadow' version "$shadowJarVersion"
-  }
+    plugins {
+        id 'com.github.johnrengelman.shadow' version "8.1.1"
```

Comment:
why not in version catalog?

Reply:
because here we get dilemma "which comes first versions or versions preprocessing" :D